File: wrappers/mvc/src/Kendo.Mvc/UI/Splitter/SplitterPane.cs
namespace Kendo.Mvc.UI
{
    using System;
    using System.Collections.Generic;
    using System.Web.Routing;
    using Kendo.Mvc.Infrastructure;

    public class SplitterPane : IAsyncContentContainer, IHtmlAttributesContainer
    {
        public SplitterPane()
        {
            Scrollable = true;
            Resizable = true;

            HtmlAttributes = new RouteValueDictionary();

            Template = new HtmlTemplate();
        }

        /// <summary>
        /// Specifies the size of the pane
        /// </summary>
        public string Size
        {
            get;
            set;
        }

        /// <summary>
        /// Specifies the minimum size of the pane
        /// </summary>
        public string MinSize
        {
            get;
            set;
        }

        /// <summary>
        /// Specifies the maximum size of the pane
        /// </summary>
        public string MaxSize
        {
            get;
            set;
        }

        /// <summary>
        /// Specifies whether the pane is initially collapsed
        /// </summary>
        public bool Collapsed
        {
            get;
            set;
        }

        /// <summary>
        /// Specifies whether the pane can be collapsed by the user
        /// </summary>
        public bool Collapsible
        {
            get;
            set;
        }

        /// <summary>
        /// Specifies whether the pane can be resized by the user
        /// </summary>
        public bool Resizable
        {
            get;
            set;
        }

        /// <summary>
        /// Specifies whether the pane shows a scrollbar when its content overflows
        /// </summary>
        public bool Scrollable
        {
            get;
            set;
        }

        private string loadContentFromUrl;

        /// <summary>
        /// Specifies URL from which to load the pane content
        /// </summary>
        public string ContentUrl
        {
            get
            {
                return loadContentFromUrl;
            }

            set
            {

                loadContentFromUrl = value;
            }
        }

        /// <summary>
        /// Specifies HTML attributes for the pane
        /// </summary>
        public IDictionary<string, object> HtmlAttributes
        {
            get;
            private set;
        }

        /// <summary>
        /// Specifies the pane contents
        /// </summary>
        public HtmlTemplate Template
        {
            get;
            set;
        }

        internal IDictionary<string, object> Serialize()
        {
            var result = new Dictionary<string, object>();

            FluentDictionary.For(result)
                .Add("scrollable", Scrollable, true)
                .Add("resizable", Resizable, true)
                .Add("collapsible", Collapsible, false)
                .Add("collapsed", Collapsed, false)
                .Add("size", Size, "*")
                .Add("min", MinSize, () => !string.IsNullOrEmpty(MinSize))
                .Add("max", MaxSize, () => !string.IsNullOrEmpty(MaxSize))
                .Add("contentUrl", ContentUrl, () => !string.IsNullOrEmpty(ContentUrl));

            return result;
        }
    }
}

File: wrappers/php/lib/Kendo/UI/SplitterPane.php
<?php

namespace Kendo\UI;

class SplitterPane extends \Kendo\SerializableObject {
    protected $ignore = array('content');

    private $attributes = array();

    public function attr($key, $value) {
        $this->attributes[$key] = $value;

        return $this;
    }

    protected function addAttributes(\Kendo\Html\Element $element) {
        foreach ($this->attributes as $key => $value) {
            $element->attr($key, $value);
        }
    }

    public function createElement() {
        $element = new \Kendo\Html\Element('div');

        $this->addAttributes($element);

        $content = $this->getProperty('content');

        if (gettype($content) == "string") {
            $element->html($content);
        }

        return $element;
    }

    /**
    * Specifies the HTML attributes that will be applied to the pane element
    * @param  $value
    * @return \Kendo\UI\SplitterPane
    */

//>> Properties

    /**
    * Specifies whether a pane is initially collapsed (true) or expanded (true).
    * @param boolean $value
    * @return \Kendo\UI\SplitterPane
    */
    public function collapsed($value) {
        return $this->setProperty('collapsed', $value);
    }

    /**
    * Specifies the size of a collapsed pane defined as pixels (i.e. "200px") or as a percentage (i.e. "50%"). This value must not exceed panes.max or be less then panes.min.
    * @param string $value
    * @return \Kendo\UI\SplitterPane
    */
    public function collapsedSize($value) {
        return $this->setProperty('collapsedSize', $value);
    }

    /**
    * Specifies whether a pane is collapsible (true) or not collapsible (false).
    * @param boolean $value
    * @return \Kendo\UI\SplitterPane
    */
    public function collapsible($value) {
        return $this->setProperty('collapsible', $value);
    }

    /**
    * Specifies the URL from which to load the content of a pane.
    * @param string $value
    * @return \Kendo\UI\SplitterPane
    */
    public function contentUrl($value) {
        return $this->setProperty('contentUrl', $value);
    }

    /**
    * Specifies the maximum size of a pane defined as pixels (i.e. "200px") or as a percentage (i.e. "50%"). The size of a resized pane cannot exceed the defined maximum size.
    * @param string $value
    * @return \Kendo\UI\SplitterPane
    */
    public function max($value) {
        return $this->setProperty('max', $value);
    }

    /**
    * Specifies the minimum size of a pane defined as pixels (i.e. "200px") or as a percentage (i.e. "50%"). The size of a resized pane cannot be less than the defined minimum size.
    * @param string $value
    * @return \Kendo\UI\SplitterPane
    */
    public function min($value) {
        return $this->setProperty('min', $value);
    }

    /**
    * Specifies whether a pane is resizable (true) or not resizable (false).
    * @param boolean $value
    * @return \Kendo\UI\SplitterPane
    */
    public function resizable($value) {
        return $this->setProperty('resizable', $value);
    }

    /**
    * Specifies whether a pane is scrollable (true) or not scrollable (false).
    * @param boolean $value
    * @return \Kendo\UI\SplitterPane
    */
    public function scrollable($value) {
        return $this->setProperty('scrollable', $value);
    }

    /**
    * Specifies the size of a pane defined as pixels (i.e. "200px") or as a percentage (i.e. "50%"). This value must not exceed panes.max or be less then panes.min.
    * @param string $value
    * @return \Kendo\UI\SplitterPane
    */
    public function size($value) {
        return $this->setProperty('size', $value);
    }

    /**
    * Sets the HTML content of the SplitterPane.
    * @param string $value
    * @return \Kendo\UI\SplitterPanes    */
    public function content($value) {
        return $this->setProperty('content', $value);
    }

    /**
    * Starts output bufferring. Any following markup will be set as the content of the SplitterPane.
    */
    public function startContent() {
        ob_start();
    }

    /**
    * Stops output bufferring and sets the preceding markup as the content of the SplitterPane.
    */
    public function endContent() {
        $this->content(ob_get_clean());
    }
//<< Properties
}

?>
